GameServerLib/Logic/Maps/Map.cs

```diff
@@ -26,30 +23,56 @@ public Map(Game game)
         {
             _game = game;
             Id = _game.Config.GameConfig.Map;
-            var path = System.IO.Path.Combine(
+            var path = Path.Combine(
                 Program.ExecutingDirectory,
                 "Content",
                 "Data",
                 _game.Config.ContentManager.GameModeName,
                 "AIMesh",
                 "Map" + Id,
-                "AIPath.aimesh"
+                "AIPath.aimesh_ngrid"
             );
 
             if (File.Exists(path))
             {
-                AIMesh = new RAF.AIMesh(path);
+                NavGrid = NavGridReader.ReadBinary(path);
             }
             else
             {
-                _logger.LogCoreError("Failed to load Summoner's Rift data.");
+                _logger.LogCoreError("Failed to load navigation graph. Aborting map load.");
                 return;
             }
 
             AnnouncerEvents = new List<Announce>();
             CollisionHandler = new CollisionHandler(this);
-            MapGameScript = new SummonersRift();
+            MapGameScript = GetMapScript(Id);
         }
+
+        public MapGameScript GetMapScript(int mapId)
+        {
+            var dict = new Dictionary<int, Type>
```

Comment:
Is this really necessary because i was thinking of loading ``MapGameScript`` from content folder?